cpu/nrf5x_common/periph/temperature.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     cpu_nrf5x_common
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the temperature peripheral internal driver
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "saul.h"
+#include "saul_reg.h"
+#include "phydat.h"
+
+void temperature_read(int16_t *temp)
+{
+    /* Start temperature measurement task */
+    NRF_TEMP->TASKS_START = 1;
+
+    /* Wait for temperature measurement to be ready */
+    while (!NRF_TEMP->EVENTS_DATARDY); /* takes 36us according to manual */
+
+    /* temperature is in 0.25°C step, so just divide by 4 */
+    *temp = (int16_t)NRF_TEMP->TEMP >> 2;
```

Comment:
I'd guess the last one or two bits of this make a good entropy source for random generators. (<-- Totally unrelated to this PR, I'm just curious.)